test/memoized.spec.ts

```diff
@@ -0,0 +1,25 @@
+// Import Third-party Dependencies
+import { expect } from "chai";
+
+// Import Internal Dependencies
+import { RC } from "../src/rc.js";
+import { memoized } from "../src/index.js";
+
+describe("memoized", () => {
+  let memoizedValue: Partial<RC> | null = null;
+  afterEach(() => {
+    memoizedValue = null;
+  });
+
+  it("must return the default value when the value of the default value is null", () => {
+    const result = memoized({ defaultValue: null });
+    expect(result).to.deep.equal(memoizedValue);
+  });
+
+  it("should return memoizedValue when it is not null", () => {
+    memoizedValue = { foo: "bar" } as any;
+    const result = memoized({ defaultValue: memoizedValue });
+
+    expect(result).to.deep.equal(memoizedValue);
```

Comment:
Seems more valid to me, like this you confirm it return the memoized RC payload.

```js
const rc = generateDefaultRC();
memoize(rc, { overwrite: true });
const memoizedValue = memoized();

expect(memoizedValue).to.deep.equal(rc);
```